web/src/engine/websites/Mangazure.ts

```diff
@@ -0,0 +1,116 @@
+﻿import { Tags } from '../Tags';
+import icon from './Mangazure.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchWindowScript } from '../platform/FetchProvider';
+
+type APIitem = {
+    title: string,
+    url: string;
+}
+
+type PageData = {
+    rawdata: string,
+    server: string
+}
+
+const pageScript = `
+    new Promise(resolve =>   {
+        resolve( { rawdata: document.body.innerHTML, server : $ImgHost });
+    });
+`;
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://fetch.mangazure.com';
+    public constructor() {
+        super('mangazure', 'Mangazure', 'https://www.mangazure.com', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Turkish, Tags.Source.Aggregator);
+    }
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /^https:\/\/www\.mangazure\.com\/\d{4}\/\d+\/[^/]+.html$/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const title = await FetchWindowScript<string>(new Request(url), '$bookTitle', 500);
+        return new Manga(this, provider, new URL(url).pathname, title);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const categories = ['Tamam', 'Devam', 'Tümü'];
+        const perPage = 50;
+        const mangaList : Manga[] = [];
+
+        for (const category of categories) {
+            for (let offset = 1, run = true; run; offset += perPage) {
+                const url = new URL(`/category?q=${category}&start=${offset}&max=${perPage}`, this.apiUrl);
+                const request = new Request(url, {
+                    headers: {
+                        Referer: this.URI.href,
+                        Origin: this.URI.href
+                    }
+                });
+                const results = await FetchJSON<APIitem[]>(request);
+                results.forEach(manga => mangaList.push(new Manga(this, provider, new URL(manga.url).pathname, this.cleanMangaTitle(manga.title))));
+                run = results.length > 0;
+            }
+        }
+        return mangaList.distinct();
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const slug = this.convertToSlug(manga.Title);
+        const url = new URL(`/book?q=${slug}`, this.apiUrl);
+        const request = new Request(url, {
+            headers: {
+                Referer: this.URI.href,
+                Origin: this.URI.href
```

Comment:
The origin header must not include path, search or fragments, use `this.URI.origin` instead.